src/core/sprites/webgl/SpriteRenderer.js

```diff
@@ -108,16 +109,13 @@ export default class SpriteRenderer extends ObjectRenderer
         // step 2: check the maximum number of if statements the shader can have too..
         this.MAX_TEXTURES = checkMaxIfStatmentsInShader(this.MAX_TEXTURES, gl);
 
-        this.shaders = new Array(this.MAX_TEXTURES);
-        this.shaders[0] = generateMultiTextureShader(gl, 1);
-        this.shaders[1] = generateMultiTextureShader(gl, 2);
+        const shader = this.shader = generateMultiTextureShader(gl, this.MAX_TEXTURES);
```

Comment:
yup! the dummy textures mean that we always use the maximum sized shader. This is essential as it may be that a texture ends up being stored in say slot 1 and 6.